force-app/main/tools/classes/zChatterAgentTool.cls

```diff
@@ -0,0 +1,109 @@
+public class zChatterAgentTool implements IAgentTool {
+    public String getDescription() {
+        return 'Chatter Agent Tool for retrieving, inserting, updating, upserting, and deleting Chatter posts';
```

Comment:
This is a lot of context for a single tool.  I think the agent will react much better to have all of these split out into separate tools (`RetrieveChatterPost`, `CreateChatterPost`, `DeleteChatterPost`).  

The way it current is, the agent will likely get confused how to use it.